ocaml/libs/http-lib/http.ml

```diff
@@ -194,10 +194,20 @@ let urlencode param =
   in
   fn chars
 
-(** Parses strings of the form a=b&c=d into ["a", "b"; "c", "d"] *)
-let parse_keyvalpairs xs =
+(** Parses strings of the form a=b;c=d (new, RFC-compliant cookie format)
+    and a=b&c=d (old, incorrect style) into [("a", "b"); ("c", "d")] *)
+let parse_cookies xs =
+  (* Determine if ';' or '&' is used as the separator.
+     Both are assumed to be illegal characters otherwise *)
+  let sep =
+    match Astring.String.find (fun c -> c = ';') xs with
```

Comment:
Is this a reasonable usage case? Could do it if people think it's something worth supporting.

Reply:
I would think this is a bit weird but it's a simple solution because you don't have to detect the delimiter but rather accept both. Later we we restrict this by accepting only one and failing we we detect the other.